src/transformers/pipelines/text_generation.py

```diff
@@ -229,14 +251,25 @@ def preprocess(
         max_length=None,
         **generate_kwargs,
     ):
-        inputs = self.tokenizer(
-            prefix + prompt_text,
-            return_tensors=self.framework,
-            truncation=truncation,
-            padding=padding,
-            max_length=max_length,
-            add_special_tokens=add_special_tokens,
-        )
+        if isinstance(prompt_text, Chat):
+            inputs = self.tokenizer.apply_chat_template(
+                prompt_text.messages,
+                padding=padding,
+                add_generation_prompt=True,
+                return_tensors=self.framework,
+                max_length=max_length,
+                truncation=truncation,
+                return_dict=True,
+            )
+        else:
+            inputs = self.tokenizer(
+                prefix + prompt_text,
+                return_tensors=self.framework,
+                truncation=truncation,
+                padding=padding,
+                max_length=max_length,
+                add_special_tokens=add_special_tokens,
+            )
```

Comment:
nit - same here (+ my own personal preference for `return_tensors` being the last option passed)

```suggestion
            inputs = self.tokenizer.apply_chat_template(
                prompt_text.messages,
                truncation=truncation,
                padding=padding,
                max_length=max_length,
                add_generation_prompt=True,
                return_dict=True,
                return_tensors=self.framework,
            )
        else:
            inputs = self.tokenizer(
                prefix + prompt_text,
                truncation=truncation,
                padding=padding,
                max_length=max_length,
                add_special_tokens=add_special_tokens,
                return_tensors=self.framework,
            )
```